src/Webservice/Webservice.php

```diff
@@ -282,15 +283,15 @@ protected function _logQuery(Query $query, LoggerInterface $logger)
      * Loops through the results and turns them into resource objects
      *
      * @param array $results Array of results from the API
-     * @param string $resourceClass The resource class to use
+     * @param \Muffin\Webservice\Model\Endpoint $endpoint The endpoint class to use
      *
-     * @return Resource[] Array of resource objects
+     * @return \Muffin\Webservice\Model\Resource[] Array of resource objects
      */
-    protected function _transformResults(array $results, $resourceClass)
+    protected function _transformResults(array $results, Endpoint $endpoint)
```

Comment:
I usually prefer object passed first ahead of the rest - if we are to break the API (changing from string to object) should the signature be changed as well?